### ISSUE TITLE
Do not use StrategicMergePatch for mocking server side apply with in unit tests

### ISSUE THREAD

**Issue by mimowo** [kubernetes/kubernetes#116851]
This is the problematic code: https://github.com/kubernetes/kubernetes/blob/3cf9f66e90d560ac080687610933c712bcf37b39/staging/src/k8s.io/client-go/testing/fixture.go#L184 introduced in https://github.com/kubernetes/kubernetes/pull/113304.

As discussed: https://github.com/kubernetes/kubernetes/pull/113304#discussion_r1144111578 this might be providing misleading unit test results, covering some real life subtle bugs.

The approach is to revert the change, the unit tests which use this change should be adjusted to provide a custom reactor which handles the patches, as suggested here: https://github.com/kubernetes/kubernetes/pull/113304#discussion_r1144715681.

**Comment by liggitt** [comment]
cc @aojea

**Comment by liggitt** [comment]
opened https://github.com/kubernetes/kubernetes/pull/116855 to drop that line to flush out in-tree unit tests

**Comment by alculquicondor** [comment]
are we targeting 1.28 for this revert?

**Comment by liggitt** [comment]
yeah, it's too late to rework that many unit tests for 1.27 😞

**Comment by alculquicondor** [comment]
/assign @apelisse 
according to the other thread

**Comment by apelisse** [comment]
Thanks for creating the issue, I was working on this without having a proper bug tracking it 👍

**Comment by furkatgofurov7** [comment]
@apelisse hi 👋🏼, v1.28 Bug Triage Lead here.

Just checking-in (it is still early in the release cycle, no rush at all) if this is still on track for the 1.28 release?

**Comment by furkatgofurov7** [comment]
@mimowo @apelisse hey folks!

Just a reminder that the code freeze is starting [01:00 UTC Wednesday 19th July 2023 / 18:00 PDT Tuesday 18th July 2023](https://everytimezone.com/s/72ee8496) (about 3 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

**Comment by furkatgofurov7** [comment]
@mimowo @apelisse hi again! 

This issue has not been updated for a long time now and I'd like to check what's the status with you once again.
FYI, as I informed about it already 2 weeks ago, we are in week 9 currently and the code freeze is starting next week (week 10) at [01:00 UTC Wednesday 19th July 2023 / 18:00 PDT Tuesday 18th July 2023](https://everytimezone.com/s/72ee8496).

Please remove the milestone if this not currently tracked for v1.28 release or if it is still, please make sure that the fix is in before the code freeze date above, thanks!

**Comment by apelisse** [comment]
Yeah, feel free to remove the milestone, I'll have to postpone to next release :-(

**Comment by alexzielenski** [comment]
> 
Yeah, feel free to remove the milestone, I'll have to postpone to next release :-(

/cc @apelisse 
/triage accepted

Now that the release cycle is over do you plan on coming back to this?

**Comment by apelisse** [comment]
I guess i'll work on this for the next release :-(
